### PR TITLE
Feature/user percentile update

### PR DESCRIPTION
**Pull request by julachancey**
### Summary

a user's percentiles will now be stored in the user schema and it will be recalculated any time they login or redeem a point. This is important because the percentiles were being calculated every time the get User query was called which was very cost ineffective. With this modified way, the individual user's calculation will only be performed during login and point redemption which is more cost efficient and should allow for less time loading and a more seamless user experience.

### Reference

_Reference your Asana task here as shown below_

[Asana link](https://app.asana.com/0/1202843173947642/1203559236062220)

### Extra

@yairtemkin

### CODE REVIEW

File: graphql/resolvers/users.js [hunk 1]
```diff
@@ -610,10 +672,96 @@ module.exports = {
...
       }
     },
 
+    async resetFallPercentile(_) {
+      const users = await User.find();
+
+      for (let i = 0; i < users.length; i++) {
+        var percentileUpdate = {};
+
+        percentileUpdate = {
```
Comment (yairtemkin) [inline graphql/resolvers/users.js:687]: Move this outside of the for loop

**Review by yairtemkin — COMMENTED** [review]
Please add more details to your summary, explain why this feature is important and what specific things did you change. It looks good for the most part, I'd simply want to find a way to test it with Omar's fake database.

File: index.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ require("dotenv").config();
 
 const resolvers = require("./graphql/resolvers");
 
-const port = process.env.PORT || 5000;
+const port = process.env.PORT || 6000;
```
[the remark this answers is not in the document]
Comment (julachancey) [reply index.js:16]: forgot to change it back

**Review by yairtemkin — APPROVED** [review]
Looks so good! This update is gonna be really effective. Good job Julia 🥳 🥳 🥳

**Pull request merged**